src/Core/Foundation/Version/Version.php

```diff
@@ -0,0 +1,255 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+namespace PrestaShop\PrestaShop\Core\Foundation\Version;
+
+use PrestaShop\PrestaShop\Core\Foundation\Version\Exception\InvalidVersionException;
+
+/**
+ * Class responsible of managing the right version of Shop
+ * for every internal/external services.
+ */
+class Version
```

Comment:
@mickaelandrieu The name is still redundant `Version\Version`, why version should be an exception when we don't want to do like that for other classes :/

Reply:
@PierreRambaud As I said before I think it is a necessity for PSR-4? Is that wrong? Other question: is it kind of forbidden in PrestaShop project? In Symfony this is quite common.

Reply:
maybe `Versioning\Version` would be better?